cmd/gke-exec-auth-plugin/cache.go

```diff
@@ -46,16 +48,25 @@ func getKeyCert() ([]byte, []byte, error) {
 }
 
 func getNewKeyCert(dir string) ([]byte, []byte, error) {
-	glog.Info("generating new private key")
-	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
-	if err != nil {
-		return nil, nil, err
-	}
-	keyBytes, err := x509.MarshalECPrivateKey(key)
-	if err != nil {
-		return nil, nil, err
+	keyPEM, err := ioutil.ReadFile(filepath.Join(dir, tmpKeyFileName))
```

Comment:
we should duplicate what we do here:

https://github.com/kubernetes/kubernetes/blob/845a55dbbda9a35cfd3659f1fe880d35a03dc44b/pkg/kubelet/certificate/bootstrap/bootstrap.go#L81-L93

e.g. we need to verify the key data before sending the csr.

Reply:
Done